core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -405,4 +435,23 @@ private String fullTableName(TableIdentifier ident) {
 
     return Pair.of(tableClient, tableIO);
   }
+
+  private Map<String, String> fetchAndMergeWithServerConf(Map<String, String> props) {
```

Comment:
I think this would be cleaner if it just fetched the config and returned the response. That way it does one only thing. The only call that would be moved outside is the `merge` call, which probably shouldn't be in the `try` block anyway.

Then the caller would look like this:

```java
    ConfigResponse config = fetchConfig(properties);
    Map<String, String> props = config.merge(properties);
```

Reply:
Yeah I had already moved it to the `try` block, but I will move the merging out so that this function has more of a single purpose.

Reply:
Updated.